tools/SourceKit/include/SourceKit/Support/Concurrency.h

```diff
@@ -17,6 +17,55 @@
 
 namespace SourceKit {
 
+class Semaphore {
+public:
+  Semaphore(long count) {
+    ImplObj = Impl::create(count);
+  }
+  ~Semaphore() {
+    if (ImplObj)
+      Impl::release(ImplObj);
+  }
+
+  void signal() {
+    Impl::signal(ImplObj);
+  }
+
+  bool wait() {
+    return Impl::wait(ImplObj);
+  }
+
+  bool wait(long milliseconds) {
+    return Impl::wait(ImplObj, milliseconds);
+  }
+
+  Semaphore(const Semaphore &Other) {
```

Comment:
The move constructor LGTM.  The copy constructor you can just define as `= delete`, and that should make `Semaphore` a move-only type.

`operator=(Semaphore &&Other)` is actually OK for a move-only type.